### PR TITLE
Extend XeTile.tile to support scatter load (experimental)

### PR DESCRIPTION
**Pull request by Jianhui-Li**
Please review these guidelines to help with the review process:
- [ ] Have you provided a meaningful PR description?
- [ ] Have you added a test, a reproducer, or a reference to an issue with a reproducer?
- [ ] Have you tested your changes locally for CPU and GPU devices?
- [ ] Have you made sure that new changes do not introduce compiler warnings?
- [ ] If this PR is a work in progress, are you filing the PR as a draft?
- [ ] Have you organized your commits logically and ensured each can be built by itself?

### CODE REVIEW

File: docs/rfcs/XeTile.md [hunk 1]
```diff
@@ -208,6 +207,22 @@ With the data being presented as 4D vector, all the vector based XeTile operatio
 ```
 The tile_pack and tile_unpack operation is similar to pack and unpack operation of tensor dialect. The source vector must be a 2D dimension vector, and no permutation is allowed for the result 4D vector, so effectively the blocking effect is identical to tensor pack/unpack operation with inner_dims_pos = [0,1] inner_dims_pos = [0, 1].
 
+## support for load_gather and store_scatter (experimental)
+`load_gather` (aka. load) load data with each element's address being explictly specified. The tile is created with a base address and offset for each element to be load. The result tile has a `scatter` attribute to different it from the regular tile.
```
Comment (Garra1980) [inline docs/rfcs/XeTile.md:211]: load ->loads

File: docs/rfcs/XeTile.md [hunk 2]
```diff
@@ -208,6 +207,22 @@ With the data being presented as 4D vector, all the vector based XeTile operatio
...
+```mlir
+  %tile0 = XeTile.init_tile %base_addr, %tile_offsets:
+     i64, vector<1x256xi32> into tile<1x256xbf16, #scatter>
+```
+`load_gather` (aka. load) load data with prepared tile and mask. Attribute `padding` specifies the padding value for the out-of-boundary access. The default value is zero.
```
Comment (Garra1980) [inline docs/rfcs/XeTile.md:216]: load->loads

File: docs/rfcs/XeTile.md (hunk 1 above)
Comment (Jianhui-Li) [reply docs/rfcs/XeTile.md:211]: fixed

File: docs/rfcs/XeTile.md (hunk 2 above)
Comment (Jianhui-Li) [reply docs/rfcs/XeTile.md:216]: fixed

File: docs/rfcs/XeTile.md [hunk 3]
```diff
@@ -208,6 +207,22 @@ With the data being presented as 4D vector, all the vector based XeTile operatio
 ```
 The tile_pack and tile_unpack operation is similar to pack and unpack operation of tensor dialect. The source vector must be a 2D dimension vector, and no permutation is allowed for the result 4D vector, so effectively the blocking effect is identical to tensor pack/unpack operation with inner_dims_pos = [0,1] inner_dims_pos = [0, 1].
 
+## support for load_gather and store_scatter (experimental)
+`load_gather` (aka. load) loads data with each element's address being explictly specified. The tile is created with a base address and offset for each element to be load. The result tile has a `scatter` attribute to different it from the regular tile.
```
Comment (Garra1980) [inline docs/rfcs/XeTile.md:211]: different - > distinguish?

File: docs/rfcs/XeTile.md [hunk 4]
```diff
@@ -208,6 +207,22 @@ With the data being presented as 4D vector, all the vector based XeTile operatio
...
+```mlir
+  %tile0 = XeTile.init_tile %base_addr, %tile_offsets:
```
Comment (Garra1980) [inline docs/rfcs/XeTile.md:213]: XeTile->xetile here and below since it is a name for dialect operation

**Comment by Garra1980** [comment]
@chencha3 pls take a look, do you think this is enough for now?

File: docs/rfcs/XeTile.md (hunk 4 above)
Comment (Jianhui-Li) [reply docs/rfcs/XeTile.md]: fixed.

**Review by chencha3 — APPROVED** [review]
LGTM

**Review by Garra1980 — APPROVED** [review]

**Comment by Garra1980** [comment]
let's merge after conflicts resolved

**Pull request merged**